examples/typed-transformer/Main.hs

```diff
@@ -291,7 +291,16 @@ evaluation ::
     outputs
     target
     m.
-  ( HasScatter workerDevices dataDevice input inputs,
+  ( 1 <= seqLen,
+    1 <= numEmbeds,
+    workerDevices ~ WorkerDevices,
+    modelDevice ~ ModelDevice,
+    dtype ~ 'Float,
+    inputs ~ '[Tensor '( 'CPU, 0) 'Int64 '[batchSize, seqLen]],
+    outputs ~ '[Tensor '( 'CPU, 0) dtype '[batchSize, seqLen, numEmbeds]],
+    models ~ '[Model numEmbeds modelDevice],
+    model ~ Model numEmbeds modelDevice,
+    HasScatter workerDevices dataDevice input inputs,
```

Comment:
These codes fix errors where type inference doesn't work.
It has too many input/output constraints and is not very versatile.

It may be related to the first breaking changes in the ghc9.4.1 release notes.
https://downloads.haskell.org/ghc/9.4.1/docs/users_guide/9.4.1-notes.html

> There were previously cases around functional dependencies and injective type families where the result of type inference would depend on the order of constraints, as written in a source file. These cases are fundamentally ambiguous. While GHC previously made an arbitrary decision, it now notices the ambiguity and rejects the program. This means that some previously accepted programs are now rejected. The solution is to add a type annotation or type application to resolve the ambiguity.
> 
> This is the fix for [#18851](https://gitlab.haskell.org/ghc/ghc/issues/18851).